### DISCUSSION TITLE
Pass request context to hook callbacks

### DISCUSSION THREAD

**Discussion by destos** [piccolo-orm/piccolo_api#154] [Ideas]
Would be useful in cases where you want to conditionally change row values based on extra information in the request context.

**Comment by dantownsend** [comment]
Yeah, this was a bit of an oversight. It might be possible to add it in a backwards compatible way by checking if the hook accepts kwargs or a request argument.

```python
import inspect

signature = inspect.signature(some_hook)
if {i for i in bar.parameters.keys()}.intersection({'kwargs', 'request'}):
    some_hook(request=request)
```

**Reply by AnthonyArmour** [reply]
Created a PR for this with @destos . I hope it's suitable
